packages/venia-concept/src/components/ToastContainer/toastContainer.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import { shape, string } from 'prop-types';
+import { useToasts } from '@magento/peregrine';
+import Toast from './toast';
+import { mergeClasses } from 'src/classify';
+import defaultClasses from './toastContainer.css';
+
+const ToastContainer = props => {
+    const classes = mergeClasses(defaultClasses, props.classes);
+    const [{ toasts }] = useToasts();
+    // Given a map of toasts each with a property "timestamp", sort and display
+    // based on the timestamp.
+    const timestampMap = new Map();
+    toasts.forEach(toast => {
+        timestampMap[toast.timestamp] = toast;
+    });
+
+    const toastList = Object.keys(timestampMap)
+        .sort()
```

Comment:
I think we can cut out one pass here, and stick to iterables in the process. Maybe consider something along the lines of the following?

```jsx
const sortByTimestamp = ([, toastA], [, toastB]) => toastB.timestamp - toastA.timestamp

const toastElements = Array.from(toasts)
    .sort(sortByTimestamp)
    .map(([id, toast]) => {
        const key = toast.isDuplicate ? Math.random() : id

        return (
            <Toast key={key} {...toast} />
        )
    })
```